src/components/getData.js

```diff
@@ -0,0 +1,34 @@
+import { useState, useEffect } from 'react';
+
+function Quote() {
+  const [quote, setQuote] = useState([]);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    const fetchQuote = async () => {
+      const response = await fetch('https://api.api-ninjas.com/v1/quotes?category=happiness', {
+        headers: {
+          'X-Api-Key': 'WhmitRBUHv9LAjdRrvVfVw==QaSq4YojxGQkm1X6',
+        },
+      });
+      const fetchedQuote = await response.json();
+      setLoading(true);
+      setQuote(fetchedQuote[0].quote);
```

Comment:
- [ ] Also, kindly ensure that the fetchedQuote array has `length > 0`. If we have `length === 0`, then please show an error message. Right now, I am getting the following error in the case when a quote is not present for a specific category such as shown below👇 Let's suppose if there are no quotes present in the `happiness` category, then we will encounter the same error

```jsx
const response = await fetch('https://api.api-ninjas.com/v1/quotes?category=uasiduia', {
  headers: {
    'X-Api-Key': 'WhmitRBUHv9LAjdRrvVfVw==QaSq4YojxGQkm1X6',
  },
});
```

_solution_

```jsx
if (fetchedQuote && fetchedQuote[0]) {
  setQuote(fetchedQuote[0].quote);
} else {
  // Handler error message 
}
```
  ![image](https://github.com/sheezu900/Math-project/assets/106119105/1f8fa145-f105-43a8-ae14-518d29f61778)